model/qti/metadata/AbstractMetadataService.php

```diff
@@ -298,10 +298,29 @@ protected function getInstances($id, $interface)
         } else {
             foreach ($this->getOption($id) as $instance) {
                 if (is_a($instance, $interface, true)) {
-                    $this->instances[$id][] = new $instance();
+                    $this->instances[$id][] = $this->getInstance($instance);
                 }
             }
         }
+
         return $this->instances[$id];
     }
+
+    private function getInstance(string $instance): object
+    {
+        $isConfigurableService = is_a($instance, ConfigurableService::class, true);
+
+        if ($isConfigurableService) {
+            return $this->getServiceManager()->get($this->getConfigurableServiceKey($instance));
```

Comment:
Why are you using Service manager instead of ServiceLocator?

Reply:
As I see, there's no ServiceLocator
![image](https://user-images.githubusercontent.com/59471572/116521504-bb163b80-a8dc-11eb-9af9-5de6c22bae1f.png)